ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -156,16 +176,38 @@ export const Homepage = withUserProfile()(class extends React.Component<Props, S
     }
   }
 
+  async validateRasLoginGovLink() {
+    const code = (new URL(window.location.href)).searchParams.get('code');
+    const redirectUrl = buildRasRedirectUrl();
+    if (code) {
+      this.setState({rasLoginGovLoading: true});
+      try {
+        const profileResponse = await profileApi().linkRasAccount({
+          authCode: code,
+          redirectUrl: redirectUrl
```

Comment:
nit: ES6 allows you to avoid this duplication if the key/value variable have the same name:

```
{
  authCode: code,
  redirectUrl
});
```

Can do the same with `authCode`, if you name it accordingly above